src/components/plots/linePlot.ts

```diff
@@ -206,6 +206,11 @@ export module Plot {
       };
     }
 
+    protected _isVisibleOnPlot(pixelPoint: Point, element: Element): boolean {
+      return !(pixelPoint.x < 0 || pixelPoint.y < 0 ||
+          pixelPoint.x > this.width() || pixelPoint.y > this.height());
```

Comment:
Wouldn't a better signature be to have _isVisibleOnPlot take in an element or an SVGRect that encompasses the shape?  This would apply for points and arbitrary shapes like squares